src/index.ts

```diff
@@ -210,29 +210,42 @@ export const handleRotateKey = async (ctx: Context, _request?: Request) => {
 
 const handleClearKey = async (ctx: Context, _request?: Request) => {
 	ctx.metrics.keyClearTotal.inc();
+
+	const now = Date.now();
+
 	const keys = await ctx.bucket.ISSUANCE_KEYS.list({ shouldUseCache: false });
 
 	let latestKey = keys.objects[0];
 	const toDelete: Set<string> = new Set();
 
-	// only keep the latest key
+	const ttl = await ctx.bucket.ISSUANCE_KEYS.getTTL();
```

Comment:
the key TTL is different from the cache TTL.
It's cache+time since key issuance. For instance, at the moment, the default in this codebase would be 5h+24h, given the cron is triggered daily.

could be worth setting a static value, such as 48h, with an associated constant